Bug Report: TestVDiff2 has two hardware-sensitive assertions that flake on fast CI runners

### Overview of the Issue

`TestVDiff2` in `go/test/endtoend/vreplication/vdiff2_test.go` contains two assertions that make implicit assumptions about the speed of the host running the test. Both pass reliably on slower self-hosted runners and ~12-vCPU developer workstations, but fail intermittently (or in one case deterministically) on modern fast cloud CI runners — particularly ones with a low vCPU count but fast per-core performance.

The two affected assertions:

**1. `VDiffRowsCompared` cleanup check (`vdiff2_test.go` line 207):**

```go
vdrc, err := getDebugVar(t, statsTablet.Port, []string{"VDiffRowsCompared"})
require.NoError(t, err, ...)
require.Equal(t, "{}", vdrc, "expected VDiffRowsCompared stat to be empty but got %s", vdrc)
```

The `VDiffRowsCompared` gauge reads from `vde.controllers` on the target primary. Controller removal during workflow `Complete` (via `dropTargetVReplicationStreams` → `deleteWorkflowVDiffData` → `handleDeleteAction`) is synchronous on the tablet, but the stats sampler reads through `globalStats.mu` — a different mutex. There's a brief window where the stats reader can observe a non-empty map between `Complete` returning and the controller being torn down.

**2. `--max-diff-duration` restart check (`vdiff2_test.go` line 278):**

```go
customerRestarts := gjson.Parse(stat).Get("customer").Int()
require.Greater(t, customerRestarts, int64(0), ...)
```

The test sets `--max-diff-duration=2s` and pre-generates customer rows based on a per-vCPU heuristic:

```go
perVCpuCount := int64(100000)
perSecondCount := int64(math.Min(float64(perVCpuCount*int64(runtime.NumCPU())), 1000000))
totalRowsToCreate := seconds * perSecondCount
```

On a 4-vCPU runner this produces only ~800K rows. If the host can diff that much in < 2s — which modern fast CI runners with NVMe storage and modern CPUs readily can — the timer never fires, the customer table never restarts, and `customerRestarts > 0` fails deterministically. The 100K-rows-per-vCPU-per-second estimate was chosen for older self-hosted runners and is now too low for the fastest hosts at low vCPU counts.

### Reproduction Steps

1. Check out `main`.
2. Run `TestVDiff2` on a 4-vCPU cloud CI runner with modern CPUs (any runner where the customer diff can chew through ~800K rows in under 2 seconds).
3. Observe that `TestVDiff2/MoveTables/unsharded_to_two_shards` fails with either:
   - `expected VDiffRestartedTableDiffsCount stat to be greater than 0 for the customer table, got 0`, or
   - `expected VDiffRowsCompared stat to be empty but got {...}` (intermittent).

### Binary Version

```sh
Version: v24.0.0-rc1 (commit c14f5fb069, main branch)
```

### Operating System and Environment details

```sh
# Affected environment
Operating system: Ubuntu 22.04
Architecture: x86_64
vCPU count: 4
(Any fast cloud CI runner with modern CPUs and SSD storage.)
```

### Log Fragments

```sh
--- FAIL: TestVDiff2 (...)
    --- FAIL: TestVDiff2/MoveTables/unsharded_to_two_shards (...)
        vdiff2_test.go:278:
            Error Trace: go/test/endtoend/vreplication/vdiff2_test.go:278
            Error:       expected VDiffRestartedTableDiffsCount stat to be greater than 0 for the customer table, got 0
            Test:        TestVDiff2/MoveTables/unsharded_to_two_shards
```

### Suggested Fix

Both fixes are test-only:

1. Wrap the `VDiffRowsCompared == "{}"` assertion in `require.Eventually` with a short bounded timeout so it tolerates the brief stats-sampler/controller-cleanup race.
2. Raise `perVCpuCount` from 100K to 250K so the existing 1M/sec cap is hit at vCPU counts ≥ 4, giving ~2M rows on every host size — the same amount that currently works on higher-vCPU hosts.

I'll open a PR with both changes.

--- a/go/test/endtoend/vreplication/vdiff2_test.go
+++ b/go/test/endtoend/vreplication/vdiff2_test.go
@@ -201,10 +201,13 @@ func TestVDiff2(t *testing.T) {
 	require.Greater(t, count, 0, "expected VDiffRowsComparedTotal stat to be greater than 0 but got %d", count)
 
 	// The VDiffs should all be cleaned up so the VDiffRowsCompared value, which
-	// is produced from controller info, should be empty.
-	vdrc, err := getDebugVar(t, statsTablet.Port, []string{"VDiffRowsCompared"})
-	require.NoError(t, err, "failed to get VDiffRowsCompared stat from %s-%d tablet: %v", statsTablet.Cell, statsTablet.TabletUID, err)
-	require.Equal(t, "{}", vdrc, "expected VDiffRowsCompared stat to be empty but got %s", vdrc)
+	// is produced from controller info, should be empty. Controller cleanup on
+	// workflow completion can briefly lag the stats sampler, so poll.
+	require.Eventually(t, func() bool {
+		vdrc, err := getDebugVar(t, statsTablet.Port, []string{"VDiffRowsCompared"})
+		require.NoError(t, err, "failed to get VDiffRowsCompared stat from %s-%d tablet: %v", statsTablet.Cell, statsTablet.TabletUID, err)
+		return vdrc == "{}"
+	}, 30*time.Second, 500*time.Millisecond, "expected VDiffRowsCompared stat on %s-%d tablet to eventually be empty", statsTablet.Cell, statsTablet.TabletUID)
 }
 
 func testWorkflow(t *testing.T, vc *VitessCluster, tc *testCase, tks *Keyspace, cells []*Cell) {
@@ -254,9 +257,10 @@ func testWorkflow(t *testing.T, vc *VitessCluster, tc *testCase, tks *Keyspace, 
 		seconds := int64(dur.Seconds())
 		chunkSize := int64(100000)
 		// Take the test host/runner vCPU count into account when generating rows.
-		perVCpuCount := int64(100000)
-		// Cap it at 1M rows per second so that we will create betweeen 100,000 and 1,000,000
-		// rows for each second in the diff duration, depending on the test host vCPU count.
+		// Raise the per-vCPU estimate so that low-vCPU but fast cloud runners (e.g.
+		// Depot 4-vCPU) still hit the 1M/sec cap and generate enough rows to force
+		// a table diff restart within --max-diff-duration.
+		perVCpuCount := int64(250000)
 		perSecondCount := int64(math.Min(float64(perVCpuCount*int64(runtime.NumCPU())), 1000000))
 		totalRowsToCreate := seconds * perSecondCount
 		log.Info(fmt.Sprintf("Test host has %d vCPUs. Generating %d rows in the customer table to test --max-diff-duration", runtime.NumCPU(), totalRowsToCreate))

